pkg/component/gardener/resourcemanager/resource_manager.go

```diff
@@ -1123,29 +1123,26 @@ func (r *resourceManager) emptyPodDisruptionBudget() *policyv1.PodDisruptionBudg
 	}
 }
 
+func (r *resourceManager) getPrometheusLabel() string {
+	if r.values.TargetDiffersFromSourceCluster {
+		return shoot.Label
+	}
+	return seed.Label
+}
```

Comment:
I don't think it makes much sense for now since it (a) depends on the values of the component (GRM uses `TargetDiffersFromSourceCluster`), others might use something else, and (b) it might not always be the `seed.Label` but could also be the `aggregate.Label`.